homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,421 @@
+    # <homeassistant.config_entries.ConfigEntry object at 0x7ffa892987d0>
+    # print(f"entry.entry_id= {entry.entry_id}") #b66d66f250b9c6f25d9cd488630dacee
+    # print(f"hass.data[DOMAIN][entry.entry_id]= {hass.data[DOMAIN][entry.entry_id]}")
+    # {'lektrico_client': Charger(_host='192.168.100.11', request_timeout=8,
+    # session=<aiohttp.client.ClientSession object at 0x7ffa67284760>,
+    # _close_session=False)}
+    # print(f"hass.data[DOMAIN][entry.entry_id][DATA_LEKTRICO_CLIENT]=
+    # {hass.data[DOMAIN][entry.entry_id][DATA_LEKTRICO_CLIENT]}")
+    # Charger(_host='192.168.100.11', request_timeout=8,
+    # session=<aiohttp.client.ClientSession object at 0x7ffa67284760>,
+    # _close_session=False)
+
+    print(f"entry.title={entry.title}")  # entry.title=lek
+    print(f"entry.data={entry.data}")
+    # entry.data={'host': '192.168.100.11', 'friendly_name': 'lek'}
+    print(f"entry.data.friendly_name={entry.data[CONF_FRIENDLY_NAME]}")
+    # entry.data.friendly_name=lek
+
+    await charger.charger_info()
+    settings = await charger.charger_config()
+    _lektrico_device = LektricoDevice(charger, hass, entry.data[CONF_FRIENDLY_NAME])
+    if not await _lektrico_device.init_device():
+        _LOGGER.error("Error initializing Lektrico Device. Name: 1P7K")
+
+    sensors = []
+
+    sensors.extend(
+        [
+            LektricoSensor(
+                charger,
+                sensor_desc,
+                settings,
+                _lektrico_device,
+                entry.data[CONF_FRIENDLY_NAME],
+            )
+            for sensor_desc in SENSORS
+        ]
+    )
+
+    async_add_entities(sensors, True)
+
+    # platform = async_get_current_platform()
+    # platform.async_register_entity_service(
+    #     SERVICE_IDENTIFY,
+    #     {},
+    #     LektricoSensor.async_identify.__name__,
+    # )
+
+
+# def get_entity_name(device, ent_key, ent_name) -> str:
+#     """Get the name for the entity"""
+#     name_slug = device.name
+#     # if ent_key == DEFAULT_SENSOR:
+#     #     return name_slug
+
+#     name = ent_name or ent_key
+#     if not ent_name:
+#         feat_name = device.available_features.get(ent_key)
+#         if feat_name and feat_name != ent_key:
+#             name = feat_name.replace("_", " ").capitalize()
+
+#     return f"{name_slug} {name}"
+
+
+class LektricoSensor(CoordinatorEntity, SensorEntity):
+    """The entity class for Lektrico charging stations sensors."""
+
+    entity_description: LektricoSensorEntityDescription
+
+    def __init__(
+        self,
+        charger: lektricowifi.Charger,
+        description: LektricoSensorEntityDescription,
+        settings: lektricowifi.Settings,
+        _lektrico_device: LektricoDevice,
+        friendly_name: str,
+    ) -> None:
+        """Initialize Lektrico charger."""
+        if _lektrico_device.coordinator is None:
+            super()
+        else:
+            super().__init__(_lektrico_device.coordinator)
```

Comment:
It should always have a coordinator? How would it be possible to not have the coordinator?

Reply:
I removed the checks for coordinator is none, thank you.